src/main/java/wooteco/subway/line/LineService.java

```diff
@@ -0,0 +1,64 @@
+package wooteco.subway.line;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional
+public class LineService {
+
+    private final LineDao lineDao;
+
+    public LineService(final LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    public Line create(final Line line) {
+        validateName(line.getName());
+
+        final Long id = lineDao.save(line.getName(), line.getColor());
+        return findById(id);
+    }
+
+    public void update(final Line line) {
+        final Long id = line.getId();
+
+        validateExisting(id);
+
+        final String newName = line.getName();
+        final String oldName = findById(id).getName();
+
+        if(!oldName.equals(newName)){
+            validateName(newName);
+        }
+
+        lineDao.update(line.getId(), line.getName(), line.getColor());
+    }
+
+    public void delete(final Long id) {
+        validateExisting(id);
+
+        lineDao.delete(id);
```

Comment:
> 어떠신가요?
닉이 보셨을 때는 지금 제 코드의 validateExisting()이 의미가 있는 메서드 정의라고 느껴지시나요?
> 
> 닉이라면 어떻게 id 존재 여부 확인을 하셨을지가 궁금합니다.

```suggestion
    public void delete(final Long id) {
        Optional<Line> optionalLine = lineDao.findById(id);
        if (optionalLine.isPresent()) {
            lineDao.delete(id);
            return;
        }
        
        throw new IllegalArgumentException("존재하지 않는 노선입니다.");
```

어차피 id 단건 조회 메서드가 Optional 반환이니, 이런 방식은 어떨까요? :)

Reply:
```java
public void delete(final Long id) {
        final Optional<Line> optionalLine = lineDao.findById(id);
        
        optionalLine.ifPresent((line)->{
            deleteAllSectionsInLine(id);
            lineDao.delete(id);
        });
    }
```

이렇게 수정해보았습니다.

감사합니다.